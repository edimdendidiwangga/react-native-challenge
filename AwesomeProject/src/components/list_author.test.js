import React from 'react';
import ListAuthor from './list_author';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import store from '../store'

const articles = [{
  urlToImage : 'www',
  author : 'edim',
  title : 'react native',
  publishedAt : '2017-09-26',
  description: 'bla bla'
},
{
  urlToImage : 'www',
  author : 'edim',
  title : 'react native',
  publishedAt : '2017-09-26',
  description: 'bla bla'
},
{
  urlToImage : 'www',
  author : 'edim',
  title : 'react native',
  publishedAt : '2017-09-26',
  description: 'bla bla'
}]

beforeEach(function() {
  store.dispatch({
    type: 'FETCH_ARTICLES',
    payload: articles
  })
})

it('renders without crashing', ()=> {
  const tree = renderer.create(<Provider store={store}><ListAuthor/></Provider>)
  expect(tree).toMatchSnapshot()
})
