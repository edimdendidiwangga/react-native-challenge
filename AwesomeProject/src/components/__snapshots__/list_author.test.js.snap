// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
      "paddingTop": 55,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          avatar={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 15,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
              }
            }
            virtual={undefined}
          >
            <Image
              source={
                Object {
                  "uri": "www",
                }
              }
              style={
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                }
              }
              virtual={undefined}
            />
          </View>
          <View
            style={
              Object {
                "alignItems": null,
                "alignSelf": null,
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 1,
                "marginLeft": 15,
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              virtual={undefined}
            >
              edim
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              virtual={undefined}
            >
              Hacker News
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "alignSelf": "stretch",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 0,
                "paddingBottom": 10,
                "paddingLeft": null,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "alignSelf": null,
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "200",
                }
              }
              virtual={undefined}
            >
              3:43 pm
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          avatar={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 15,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
              }
            }
            virtual={undefined}
          >
            <Image
              source={
                Object {
                  "uri": "www",
                }
              }
              style={
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                }
              }
              virtual={undefined}
            />
          </View>
          <View
            style={
              Object {
                "alignItems": null,
                "alignSelf": null,
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 1,
                "marginLeft": 15,
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              virtual={undefined}
            >
              edim
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              virtual={undefined}
            >
              Hacker News
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "alignSelf": "stretch",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 0,
                "paddingBottom": 10,
                "paddingLeft": null,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "alignSelf": null,
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "200",
                }
              }
              virtual={undefined}
            >
              3:43 pm
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          avatar={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": undefined,
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 15,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
              }
            }
            virtual={undefined}
          >
            <Image
              source={
                Object {
                  "uri": "www",
                }
              }
              style={
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                }
              }
              virtual={undefined}
            />
          </View>
          <View
            style={
              Object {
                "alignItems": null,
                "alignSelf": null,
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 1,
                "marginLeft": 15,
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              virtual={undefined}
            >
              edim
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              virtual={undefined}
            >
              Hacker News
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "alignSelf": "stretch",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 0,
                "paddingBottom": 10,
                "paddingLeft": null,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "alignSelf": null,
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "200",
                }
              }
              virtual={undefined}
            >
              3:43 pm
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={null}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 20,
        }
      }
    >
      <View
        danger={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.8)",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        success={false}
        virtual={undefined}
        warning={false}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "flex": 1,
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#757575",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          virtual={undefined}
        />
        <RCTScrollView
          dataArray={Array []}
          dataSource={
            ListViewDataSource {
              "items": 0,
            }
          }
          enableEmptySections={true}
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 15,
            }
          }
          virtual={undefined}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
